tests/framework/utils_cpu_templates.py

```diff
@@ -23,16 +23,18 @@ def get_supported_cpu_templates():
     """
     Return the list of CPU templates supported by the platform.
     """
+    # pylint:disable=too-many-return-statements
     match cpuid_utils.get_cpu_vendor():
```

Comment:
Nit: You can use [structured matching](https://benhoyt.com/writings/python-pattern-matching/) to refactor this. I think an implementation like below is more readable.
```python
vendor = cpuid_utils.get_cpu_vendor()
cpu_model = global_props.cpu_model
host_linux = global_props.host_linux_version_tpl
cpu_codename = global_props.cpu_codename

match (vendor, cpu_codename, cpu_model):
    # T2CL template is only supported on Cascade Lake and newer CPUs.
    case (cpuid_utils.CpuVendor.INTEL, cpuid_utils.CpuModel.INTEL_SKYLAKE, _):
        return sorted(set(INTEL_TEMPLATES) - set(["T2CL"]))
    case (cpuid_utils.CpuVendor.INTEL, _, _):
        return INTEL_TEMPLATES
    case (cpuid_utils.CpuVendor.AMD, _, _):
        return AMD_TEMPLATES
    case (cpuid_utils.CpuVendor.ARM, _, cpuid_utils.CpuModel.ARM_NEOVERSE_V1) if host_linux >= (6,1):
        return ARM_TEMPLATES
return []
```

Reply:
Never mind I will split this into a separate PR on top.

Reply:
I've created https://github.com/firecracker-microvm/firecracker/pull/4407